addons/resource/models/resource.py

```diff
@@ -291,7 +291,7 @@ def _get_previous_work_day(self, day_date):
         return day_date + relativedelta(days=days)
 
     @api.multi
-    def _get_leave_intervals(self, resource_id=None, start_datetime=None, end_datetime=None):
+    def _get_leave_intervals(self, resource_id=None, start_datetime=None, end_datetime=None, time_type='leave'):
```

Comment:
I would rather add an additional domain parameter. That way callers will be able to choose how to fine-tune the search on leaves.